File: DemoApp Objective-C/CASSample/InterstitialAd/InterstitialVC.m
//
//  InterstitialVC.m
//  CASSample
//

#import "AppDelegate.h"
#import "InterstitialVC.h"

@interface InterstitialVC ()

@property (strong, nonatomic) CASInterstitial *interstitial;

@end

@implementation InterstitialVC

- (void)viewDidLoad {
    [super viewDidLoad];

    self.interstitial = [[CASInterstitial alloc] initWithCasID: [AppDelegate casId]];
    self.interstitial.delegate = self;
    self.interstitial.impressionDelegate = self;
    
    self.navigationItem.largeTitleDisplayMode = UINavigationItemLargeTitleDisplayModeNever;
    self.navigationItem.title = @"Interstitial Ad";
}

- (IBAction)loadAction:(UIButton *)sender {
    [self.interstitial loadAd];
}

- (IBAction)showAction:(UIButton *)sender {
    [self.interstitial presentFromViewController:self];
}

#pragma mark - CASImpressionDelegate

- (void)adDidRecordImpressionWithInfo:(CASContentInfo *)info {
    NSLog(@"adDidRecordImpressionWithInfo");
}


#pragma mark - CASScreenContentDelegate

- (void)screenAdDidLoadContent:(id<CASScreenContent>)ad {
    NSLog(@"screenAdDidLoadContent");
}

- (void)screenAd:(id<CASScreenContent>)ad didFailToLoadWithError:(CASError *)error {
    NSLog(@"screenAd:didFailToLoadWithError %@", error.description);
}

- (void)screenAdWillPresentContent:(id<CASScreenContent>)ad {
    NSLog(@"screenAdWillPresentContent");
}

- (void)screenAd:(id<CASScreenContent>)ad didFailToPresentWithError:(CASError *)error {
    NSLog(@"screenAd:didFailToPresentWithError %@", error.description);
}

- (void)screenAdDidClickContent:(id<CASScreenContent>)ad {
    NSLog(@"screenAdDidClickContent");
}

- (void)screenAdDidDismissContent:(id<CASScreenContent>)ad {
    NSLog(@"screenAdDidDismissContent");
}

@end

File: DemoApp Swift/CASSample/InterstitialAd/InterstitialVC.swift
//
//  InterstitialVC.swift
//  CASSample
//
//  Copyright © 2025 Clever Ads Solutions. All rights reserved.
//

import UIKit
import CleverAdsSolutions

class InterstitialVC: UIViewController {
            
    @IBOutlet var loadButton: UIButton!
    @IBOutlet var showButton: UIButton!
                
    private var interstitial = CASInterstitial(casID: AppDelegate.casId)
            
    override func viewDidLoad() {
        super.viewDidLoad()
        interstitial.delegate = self
        interstitial.impressionDelegate = self
        //
        navigationItem.largeTitleDisplayMode = .never
        navigationItem.title = "Interstitial Ad"
    }
    
    @IBAction private func loadAction(_ sender: UIButton) {
        interstitial.loadAd()
    }
    
    @IBAction private func showAction(_ sender: UIButton) {
        interstitial.present(from: self)
    }
}

extension InterstitialVC: CASImpressionDelegate {
    func adDidRecordImpression(info: AdContentInfo) {
        print(#function, "Ad Source: \(info.sourceName)")
    }
}

extension InterstitialVC: CASScreenContentDelegate { 
    func screenAdDidLoadContent(_ ad: any CASScreenContent) {
        print(#function)
    }

    func screenAd(_ ad: any CASScreenContent, didFailToLoadWithError error: AdError) {
        print(#function, "Error: \(error.description)")
    }

    func screenAdWillPresentContent(_ ad: any CASScreenContent) {
        print(#function)
    }

    func screenAd(_ ad: any CASScreenContent, didFailToPresentWithError error: AdError) {        
        print(#function, "Error: \(error.description)")
    }

    func screenAdDidClickContent(_ ad: any CASScreenContent) {
        print(#function)
    }

    func screenAdDidDismissContent(_ ad: any CASScreenContent) {
        print(#function)
    }
}
